### ISSUE TITLE
Can not use ExpressionAttributeNames and ExpressionAttributeValues in DynamoDb Query

### ISSUE THREAD

**Issue by dejansijakovic** [zio/zio-dynamodb#242]
It is impossible to set ExpressionAttributeNames and ExpressionAttributeValues in QuerySome or QueryAll.

If I'm using reserved word in DynamoDb as column name i need to set these two for query to work.

in this method you are creating QueryRequest and  you have expressionAttributeNames and expressionAttributeValues
but it is impossible to set these in QuerySome or QueryAll.

 ```
 private def awsQueryRequest(querySome: QuerySome): QueryRequest = {
    val (aliasMap, (maybeFilterExpr, maybeKeyExpr, projections)) = (for {
      filter      <- AliasMapRender.collectAll(querySome.filterExpression.map(_.render))
      keyExpr     <- AliasMapRender.collectAll(querySome.keyConditionExpression.map(_.render))
      projections <- AliasMapRender.forEach(querySome.projections)
    } yield (filter, keyExpr, projections)).execute

    QueryRequest(
      tableName = ZIOAwsTableName(querySome.tableName.value),
      indexName = querySome.indexName.map(_.value).map(ZIOAwsIndexName(_)),
      select = querySome.select.map(awsSelect),
      limit = Some(querySome.limit).map(PositiveIntegerObject(_)),
      consistentRead = Some(toBoolean(querySome.consistency)).map(ZIOAwsConsistentRead(_)),
      scanIndexForward = Some(querySome.ascending),
      exclusiveStartKey = querySome.exclusiveStartKey.map(m => awsAttributeValueMap(m.map)),
      returnConsumedCapacity = Some(awsConsumedCapacity(querySome.capacity)),
      projectionExpression = toOption(projections).map(_.mkString(", ")).map(ZIOAwsProjectionExpression(_)),
      filterExpression = maybeFilterExpr.map(ZIOAwsConditionExpression(_)),
      expressionAttributeValues = aliasMapToExpressionZIOAwsAttributeValues(aliasMap),
      expressionAttributeNames = aliasMapToExpressionZIOAwsAttributeNames(aliasMap),
      keyConditionExpression = maybeKeyExpr.map(ZIOAwsKeyExpression(_))
    )
  }
```

Is there some workaround for this ?

Thanks!

**Comment by googley42** [comment]
hi, ExpressionAttributeNames and ExpressionAttributeValues are intended to be set automatically for names and values, thats why there is no entry point for doing this manually. There is an integration test for this behaviour for [`querySome` here](https://github.com/zio/zio-dynamodb/blob/series/2.x/dynamodb/src/it/scala/zio/dynamodb/LiveSpec.scala#L220-L235) and [`queryAll`here](https://github.com/zio/zio-dynamodb/blob/series/2.x/dynamodb/src/it/scala/zio/dynamodb/LiveSpec.scala#L196C1-L206) 

Is it possible for you to post a reproducer/example please?

**Comment by dejansijakovic** [comment]
Hi @googley42, here is my example. I need to write this kind of query
```
DynamoDBQuery
          .querySome[ZIOStore](ZIOStore.tableName, 1)
          .whereKey(partitionKey("id") === id && SortKey("source").beginsWith(source))
          .sortOrder(false)
```

The problem is that "source" is a reserved word in DynamoDb and I get this exception
Invalid KeyConditionExpression: Attribute name is a reserved keyword; reserved keyword: source (Service: DynamoDb, Status Code: 400

So I need to set ExpressionAttributeNames and ExpressionAttributeValues like in this example
https://stackoverflow.com/questions/42425146/query-an-amazon-dynamodb-by-column-named-timestamp-a-reserved-word

**Comment by googley42** [comment]
hi @dejansijakovic thanks for the example - I have managed to reproduce this. Its is a bug in key condition expression rendering logic that does not make use of improvements introduced in a previous PR. I will have a look at this next.

**Comment by dejansijakovic** [comment]
Thanks!

**Comment by googley42** [comment]
hi @dejansijakovic, I have merged a [PR](https://github.com/zio/zio-dynamodb/pull/245) to fix your issue and the snapshot is available [here](https://oss.sonatype.org/content/repositories/snapshots/dev/zio/zio-dynamodb_2.13/0.2.10+9-4eb5f563-SNAPSHOT/) . Integration tests are [here](https://github.com/zio/zio-dynamodb/blob/series/2.x/dynamodb/src/it/scala/zio/dynamodb/LiveSpec.scala#L210-L259)

Could you please try it out by adding the below to your `build.sbt` and let me know the result - if its OK then I will cut another release. 

```
resolvers += Resolver.sonatypeRepo("snapshots")

"dev.zio" %% "zio-dynamodb" % "0.2.10+9-4eb5f563-SNAPSHOT"
```

**Comment by dejansijakovic** [comment]
Hi @googley42 thanks! Forgot to mention that I'm using Scala 3 in my project so dynamodb_3, is it possible to build dynamodb_3 snapshot ?
Thanks!

**Comment by googley42** [comment]
Hi @dejansijakovic Scala 3 snapshot gets published at the same time and can be found at https://oss.sonatype.org/content/repositories/snapshots/dev/zio/zio-dynamodb_3/0.2.10+9-4eb5f563-SNAPSHOT/

**Comment by dejansijakovic** [comment]
@googley42 It is working as expected now. Thank you! You can release new version. Sorry it took a while, my sbt didn't want to load that snapshot, had to download it and add to project manually.

**Comment by googley42** [comment]
hi @dejansijakovic, thanks for testing the snapshot - I have published version [v0.2.11](https://github.com/zio/zio-dynamodb/releases/tag/v0.2.11) and its made its way eventually to Maven Central.
